var/spack/repos/builtin/packages/catalyst/package.py

```diff
@@ -51,23 +55,62 @@ class Catalyst(CMakePackage):
     variant('python', default=False, description='Enable Python support')
     variant('essentials', default=False, description='Enable Essentials support')
     variant('extras', default=False, description='Enable Extras support')
-    variant('rendering', default=False, description='Enable Vtk Rendering support')
+    variant('rendering', default=False, description='Enable VTK Rendering support')
+    variant('expat', default=True, description="Use external expat")
 
-    depends_on('git')
+    depends_on('git', type='build')
     depends_on('mpi')
+    depends_on('expat', when='+expat')
     depends_on('python@2:2.8', when='+python')
     depends_on('mesa', when='+rendering')
-    depends_on("libx11", when='+rendering')
-    depends_on("libxt", when='+rendering')
+    depends_on('libx11', when='+rendering')
+    depends_on('libxt', when='+rendering')
     depends_on('cmake@3.3:', type='build')
 
+    @when('@5.5.0:5.5.2')
+    def patch(self):
+        """Apply the patch (it should be fixed in Paraview 5.6)
+        at the package dir to the source code in
+        root_cmakelists_dir."""
+        patch_name = 'vtkm-catalyst-pv551.patch'
+        pkg_dir = os.path.dirname(absolute_path_for_package(self))
+        patch = which("patch", required=True)
+        with working_dir(self.root_cmakelists_dir):
+            patch('-s', '-p', '1', '-i',
+                  join_path(pkg_dir, patch_name),
+                  "-d", '.')
```

Comment:
Can this whole thing be replaced with:
```python
patch('vtkm-catalyst-pv551.patch', level=1, when='@5.5.0:5.5.999')
```
?

Reply:
That certainly looks much nicer!

Reply:
yes, it certainly looks much nicer but it does not work since the catalyst source files are not downloaded from a url but extracted from the paraview sources.
In other words, the catalyst package downloads the paraview source files, calls a script inside which extracts the catalyst files and put them in another folder. I need to apply the patch to catalyst, not to paraview. If I use "patch('vtkm-catalyst-pv551.patch', level=1, when='@5.5.0:5.5.999')", I apply the patch to the paraview source files. I tried to modify self.stage.path pointing this variable to the catalyst source files but I have had no success.
This code snippet is clearly a workaround.